draft-ietf-taps-interface.md

```diff
@@ -777,6 +777,20 @@ by using the following call on the Connection Object:
 TransportProperties := Connection.GetTransportProperties()
 ~~~
 
+An individual property is then queried via the following call on the
+`TransportProperties` object:
+
+~~~
+value := TransportProperties.Get(property)
+~~~
+
+If the given Transport Property is of type Boolean, `Has` may be provided as an
+alias for `Get`, useful for readability in such constructions as the following:
+
+~~~
+if TransportProperties.Has(boolean_property) then ...
+~~~
+
```

Comment:
Isn't this redundant (in most languages)? Wouldn't `if TransportProperties.Get(boolean_property) then` deliver the same result? In any case, this should be moved to Appendix A (convenience functions).

Reply:
True, it might not be best to have different behaviour depending on wether the connection has been established or not. Perhaps it would be an option to be able to query the transport properties on a Connection object (i.e. Connection.get_property(property)) which will return what the property actually turned out to be while queries on the TransportProperties will always return the preference level. 
I am also unsure what the current PR text would mean for reusability of TransportProperties, if the preference level is overwritten with a bool after connection establishment I guess the same TransportProperties object could not be used in a second Preconnection.

Reply:
Ooh... this might actually make the case for Get != Has: Get returns the original preference, while Has is defined only after protocol selection (e.g., throws an exception prior) and always flattens it to a boolean based on the selected protocol.

Reply:
I still think that would cause some problems if you reuse transport properties. What if you use the same TransportProperties object in two different Preconnections that lead to two Connections with different selected protocols?

Reply:
That's solvable: Just make ```has``` a method on the connection

Reply:
I think we need to discuss this one.

Reply:
I suggest adding a reference to Section 7 here, where it explains how to query the "result" of your Selection Properties on a Connection.

Reply:
* Remove getters: Get, Has, Unset are not a required part of the API for a preconnection-phase Property object.
* Clean up getting preferences in 4.2.2.